frontend_v2/src/app/components/challenge/challengeleaderboard/challengeleaderboard.component.ts

```diff
@@ -335,7 +335,7 @@ export class ChallengeleaderboardComponent implements OnInit, AfterViewInit, OnD
         SELF.router.navigate(['../../' + phaseSplit['id']], { relativeTo: this.route });
       }
       SELF.selectedPhaseSplit = phaseSplit;
-      if (SELF.selectedPhaseSplit) {
+      if (SELF.selectedPhaseSplit && !SELF.router.url.endsWith('leaderboard')) {
```

Comment:
@burnerlee this check is to make sure the leaderboard is fetched only when the user goes to the leaderboard page right? I don't understand why it is `!` of routes ending with `leaderboard`. Can you please clarify

Reply:
@Ram81 yeah sure..  What happens currently is that initially I go to the `/leaderboard` page. When I select a phase split, the function `phaseSplitSelected` is called. In this function, we start a poller to start fetching the leaderboard data (let's call this poller1) and then we redirect ourselves to `/leaderboard/phaseSplitId` where the same page is rendered and `phaseSplitSelected` is called again. Therefore this starts another poller (poller2). Now, if you see the first GIF attached you would see that there are 2 requests polled every 5 seconds currently before the fix. This is due to both the pollers. And then when we leave this page (`/leaderboard/phaseSplitId`) the poller for this page (poller2) is stopped. But poller1 is never stopped and it keeps on fetching leaderboard data. So I have added a condition that a poller is only started when we are not on the leaderboard page., so that only poller2 in the above case is started. Hence, in the second GIF attached, you can see that only 1 request is made for each poll and also it stops as soon as we exit that page

Reply:
@burnerlee then I think the check should be `url.contains('leaderboard')` and doesn't end with it. Did you also check if the polling is not started on any other tabs?

Reply:
@Ram81, actually we want the poller to start when url is something like `localhost:4200/challenge/20/leaderboard/80` and to not start when it is `localhost:4200/challenge/20/leaderboard`. So I feel endsWith would work here. 
Yes, I checked it. Actually for the poller to start there must be 2 conditions. A phaseSplit has to be selected in the leaderboard page and the url of the page at that time must not end with `leaderboard`

Reply:
@burnerlee Wouldn't it make more sense to check if the current tab is in focus, rather than checking the URL?
Something like [this](https://stackoverflow.com/questions/1760250/how-to-tell-if-browser-tab-is-active)

Reply:
@shauryaag actually the tab is active in both the conditions 😅 maybe you can try to reproduce the issue by running in your pc.. you'll get an insight on what's causing the multiple requests